sdk/core/Azure.Core.TestFramework/src/RetryOnErrorAttribute.cs

```diff
@@ -108,8 +108,8 @@ public override TestResult Execute(TestExecutionContext context)
             private bool ShouldRetry(TestExecutionContext context)
             {
                 return
-                    context.CurrentResult.ResultState.Status == TestStatus.Failed
-                    && context.CurrentResult.ResultState.Label == "Error"
+                    context?.CurrentResult?.ResultState?.Status == TestStatus.Failed
+                    && context?.CurrentResult?.ResultState?.Label == "Error"
```

Comment:
Can we add a test to cover this?

Reply:
I've been thinking about this change since last night, and I've come to understand now that it is actually unnecessary. The cases where these can be null are already taken care of by NUnit, and I had just misinterpreted their documentation. I have reverted the changes in this file.

Having said that, I re-structured the tests that I already had to make things cleaner. Specifically, I was testing the child class `RetryOnInternalServerErrorAttribute` that I use in Text Analytics, and those tests were all in Azure.AI.TextAnalytics.Tests. I'm moving those tests over to Azure.Core.TestFramework.Tests instead, where I think it makes more sense. The only ones left in Azure.AI.TextAnalytics.Tests are the ones specific to the Text Analytics use-case.